spec/scry/completion_provider_spec.cr

```diff
@@ -136,8 +136,11 @@ module Scry
     context "classname completion" do
       _kind_ = CompletionItemKind::Class
       it_completes("A", %w(Array Atomic ArgumentError AtExitHandlers))
+      it_completes(" ::A", %w(Array Atomic ArgumentError AtExitHandlers))
+      it_completes(" ::", %w())
```

Comment:
why nothing? It should give every top level classes & constants

Reply:
@bew I think listing all classes would be very slow :sweat_smile: , no?

By example on Workspace symbols, if query is empty, you get this:

![screenshot_20180509_193607](https://user-images.githubusercontent.com/3067335/39846247-3932427c-53c0-11e8-8d8a-7f8457dfea51.png)

Reply:
@faustinoaq you're showing symbols here, not completion.

Slow? I don't know. 

So for now it only gives an empty list?

Reply:
I couldn't figure out why it didn't return anything :(

Reply:
No problem, I think is ok :+1:

Reply:
@laginha87 let's not block this pr as it's not so important, but please open an issue to keep track of `::` completion, and fix it one day, or decide what we do about it

Reply:
I actually wanna tackle doing a context aware comlpetion for this at some point .
for example:
```
module A
     ::
```
should complete with modules and classes inside A as well

Reply:
@bew added https://github.com/crystal-lang-tools/scry/issues/118

Reply:
@laginha87 there is no need for contextual completion, `::` completions will ALWAYS be the top level things, that's what it exists for, there is no contextual things going on for types that starts with `::`